js_modules/dagit/packages/core/src/assets/Stale.tsx

```diff
@@ -0,0 +1,127 @@
+import {Colors, Box, BaseTag, Tooltip, Icon, Body} from '@dagster-io/ui';
+import groupBy from 'lodash/groupBy';
+import isEqual from 'lodash/isEqual';
+import React from 'react';
+
+import {displayNameForAssetKey, LiveDataForNode} from '../asset-graph/Utils';
+import {AssetKeyInput, StaleCauseCategory, StaleStatus} from '../graphql/types';
+
+export const isAssetMissing = (liveData?: LiveDataForNode) =>
+  liveData && liveData.staleStatus === StaleStatus.MISSING;
+
+export const isAssetStale = (liveData?: LiveDataForNode) =>
+  liveData && liveData.staleStatus === StaleStatus.STALE;
+
+const NO_CAUSES = 'No reasons available.';
+
+const LABELS = {
+  self: {
+    [StaleCauseCategory.CODE]: 'Code version',
+    [StaleCauseCategory.DATA]: 'Data version',
+    [StaleCauseCategory.DEPENDENCIES]: 'Dependencies',
+  },
+  upstream: {
+    [StaleCauseCategory.CODE]: 'Upstream code version',
+    [StaleCauseCategory.DATA]: 'Upstream data version',
+    [StaleCauseCategory.DEPENDENCIES]: 'Upstream dependencies',
+  },
+};
+
+export const StaleReasonsLabel: React.FC<{
+  assetKey: AssetKeyInput;
+  include: 'all' | 'upstream' | 'self';
+  liveData?: LiveDataForNode;
+  onClick?: () => void;
+}> = ({liveData, include, assetKey, onClick}) => {
+  if (!isAssetStale(liveData) || !liveData?.staleCauses.length) {
+    return null;
+  }
+
+  return (
+    <Body color={Colors.Yellow700}>
+      <Tooltip position="top" content={<StaleCausesSummary causes={liveData.staleCauses} />}>
+        <Box onClick={onClick} flex={{gap: 4}}>
```

Comment:
I don't love `onClick`s on divs. Can this be a `ButtonLink` or something instead? Also it has a `gap` applied but there's just the one string child.